src/main/java/StackADT.java

```diff
@@ -0,0 +1,10 @@
+public interface StackADT <T> {
```

Comment:
An interface is meaningless without documentation! What is `pop`? Can I just always return `null`? Or maybe I'll do FIFO? Or return a randomly initialized object.

Javadoc is your friend here. [Stack](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Stack.html) shouldn't really be used any more, but it has decent docs. [Deque](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Deque.html) is the replacement (double-ended queue; a stack and a queue).